### ISSUE TITLE
Bean validator order issue

### ISSUE THREAD

**Issue by draco1023** [springfox/springfox#2210]
A model property looks like below.
```java
@Pattern(regexp = "\\d{11}")
@NotNull
@ApiModelProperty(value = "Mobile number")
private String mobile;
```
`@NotNull` means this property must be required, but it's not in swagger ui.

So I debug into this
![image](https://user-images.githubusercontent.com/1821583/35341533-de479dbe-0160-11e8-999f-aed665d7ce8f.png)

Yes, the `NotNullAnnotationPlugin` set `required` to true, but then `ApiModelPropertyPropertyBuilder` will set it to false, since there is no `required` declaration in `@ApiModelProperty`.

Adding `required = true` to `@ApiModelProperty` works. Is there any workaround without doing this?

**Comment by MaksimOrlov** [comment]
@draco1023
That is related to https://github.com/springfox/springfox/issues/1231

**Comment by draco1023** [comment]
In my opinion, multiple validators should share information among them.
For example, if any validator set `required` to true, this property should be required;
if multiple validators set `description`, descriptions should be joined;
if multiple validators set `min` or `max`, the smallest range should be set;
...

**Comment by MaksimOrlov** [comment]
@draco1023, I agree with you. And we will definitely touch that when will be implementing JSR-303 validation groups. But that is controversial as well, according to the project goals:

> Philosophically, we want to discourage using (swagger-core) annotations that are not material to the service description at runtime. For e.g. the jackson annotations should always trump or have more weight than @ApiModelProperty or for e.g. @NotNull or specifying @RequestParam#required should always win. Annotations are to to be used only to supplement documentation or override/tweak the resulting spec in cases where its not possible to infer service/schema characteristics.

Swagger annotations were designed to **override** property definitions.

**Comment by MaksimOrlov** [comment]
As a fast solution you can create your own annotation f.e. *@Description* and own plugin to read it.

**Comment by MaksimOrlov** [comment]
Then you can use it inserted of *@ApiModelProperty*

```
@Pattern(regexp = "\\d{11}")
@NotNull
@Description(value = "Mobile number")
private String mobile;
```

**Comment by draco1023** [comment]
Thanks @MaksimOrlov . I'll try.

**Comment by dilipkrish** [comment]
@draco1023 Yeah you've hit a contentious issue :) What @MaksimOrlov suggested should absolutely work and is recommended.
